src/Console/Plugin/UninstallCommand.php

```diff
@@ -0,0 +1,235 @@
+            $plugin = new Plugin();
+            $plugin->checkPluginState($directory); // Be sure that plugin information are up to date in DB
+
+            if (!$this->canRunUninstallMethod($directory)) {
+                $failed = true;
+                continue;
+            }
+
+            if (!$plugin->getFromDBByCrit(['directory' => $directory])) {
+                $this->output->writeln(
+                    '<error>' . sprintf(__('Unable to load plugin "%s" information.'), $directory) . '</error>',
+                    OutputInterface::VERBOSITY_QUIET
+                );
+                $failed = true;
+                continue;
+            }
+            $plugin->uninstall($plugin->fields['id']);
+
+            // Check state after uninstallation
+            if ($plugin->fields['state'] != Plugin::NOTINSTALLED) {
+                $this->output->writeln(
+                    '<error>' . sprintf(__('Plugin "%s" uninstallation failed.'), $directory) . '</error>',
+                    OutputInterface::VERBOSITY_QUIET
+                );
+                $this->outputSessionBufferedMessages([WARNING, ERROR]);
+                $failed = true;
+                continue;
+            }
+
+            $message = __('Plugin "%1$s" has been uninstalled.');
+
+            $output->writeln(
+                '<info>' . sprintf($message, $directory) . '</info>',
+                OutputInterface::VERBOSITY_NORMAL
+            );
+        }
+
+        return $failed ? Command::FAILURE : Command::SUCCESS;
+    }
+
+    protected function getDirectoryChoiceQuestion()
+    {
+        return __('Which plugin(s) do you want to uninstall (comma separated values)?');
+    }
+
+    protected function getDirectoryChoiceChoices()
+    {
+        // Fetch directory list
+        $directories = [];
+        foreach (PLUGINS_DIRECTORIES as $plugins_directory) {
+            $directory_handle  = opendir($plugins_directory);
+            while (false !== ($filename = readdir($directory_handle))) {
+                if (
+                    !in_array($filename, ['.svn', '.', '..'])
+                    && is_dir($plugins_directory . DIRECTORY_SEPARATOR . $filename)
+                ) {
+                    $directories[] = $filename;
+                }
+            }
+        }
+
+        // Fetch plugins information
+        $choices = [];
+        foreach ($directories as $directory) {
+            $plugin = new Plugin();
+            $informations = $plugin->getInformationsFromDirectory($directory);
+
+            if (empty($informations)) {
+                continue; // Ignore directory if not able to load plugin information.
+            }
+
+            if ($this->isInstalled($directory)) {
+                continue;
+            }
+
+            $choices[$directory] = array_key_exists('name', $informations)
+                ? $informations['name']
+                : $directory;
+        }
+
+        ksort($choices, SORT_STRING);
+
+        return $choices;
```

Comment:
According to `Plugin` class logic, all plugins with `self::ACTIVATED, self::NOTUPDATED, self::TOBECONFIGURED, self::NOTACTIVATED` states could be uninstalled (this is the condition to display the uninstall button in UI).
Method could be simplified, see `getDirectoryChoiceChoices()` from `ActivateCommand`.